### ISSUE TITLE
cmd/doc: usage()

### ISSUE THREAD

**Issue by gertcuykens** [golang/go#34976]
Fixes usage inconsistency between

/usr/local/go/src/cmd/doc/main.go
```
func usage() {
	fmt.Fprintf(os.Stderr, "Usage of [go] doc:\n")
	fmt.Fprintf(os.Stderr, "\tgo doc\n")
	fmt.Fprintf(os.Stderr, "\tgo doc <pkg>\n")
	fmt.Fprintf(os.Stderr, "\tgo doc <pkg> <sym>[.<method>]\n")
	fmt.Fprintf(os.Stderr, "\tgo doc [<pkg>.][<sym>.]<method>\n")
	fmt.Fprintf(os.Stderr, "\tgo doc <sym>[.<method>]\n")
	fmt.Fprintf(os.Stderr, "For more information run\n")
	fmt.Fprintf(os.Stderr, "\tgo help doc\n\n")
	fmt.Fprintf(os.Stderr, "Flags:\n")
	flag.PrintDefaults()
	os.Exit(2)
}
```

/usr/local/go/src/cmd/go/internal/doc/doc.go
```
	go doc <pkg>
	go doc <pkg> <sym>[.<methodOrField>]
	go doc [<pkg>.][<sym>.]<methodOrField>
	go doc <sym>[.<methodOrField>]
```

see diff https://golang.org/cl/201857 for details
Note that the original `[<pkg>].<sym>[.<method>]` was definitely a typo `].` vs `.]` ,just trying to make it identical while i was at it fixing that.

Also note https://groups.google.com/forum/#!topic/golang-nuts/OFg-02kfpno Compared to go help doc output, it seems `flag.PrintDefaults()` messes up flags descriptions new lines of `go doc --help`? But thats probably a separate issue with `flag.PrintDefaults()`

**Comment by gertcuykens** [comment]
@julieqiu please add label `Go 1.14` and remove `needsInvestigation` before this patch gets forgotten

**Comment by bcmills** [comment]
@gertcuykens, as far as I can tell this fix doesn't need to hold up the 1.14 release if it isn't ready, so the `Go1.14` milestone probably isn't appropriate. (But it's small and cosmetic enough that it can probably make 1.14 anyway.)
